### PR TITLE
Feat change base map

### PR DESCRIPTION
**Pull request by tltk90**

### CODE REVIEW

File: src/app/@ansyn/ansyn/modules/plugins/components/imagery-change-map/imagery-change-map.component.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+				protected element: ElementRef,
+				protected logger: LoggerService,
+				@Inject(MAP_PROVIDERS_CONFIG) protected mapProvidersConfig: IMapProviderConfig) {
+	}
+
+	@AutoSubscription
+	mapTypeChange$ = () => this.store$.select(selectMapTypeById(this.mapId)).pipe(
+		tap(mapType => {
+			this.currentSourceType = this.mapProvidersConfig[mapType].defaultMapSource;
+			this.mapSources = Object.values(this.mapProvidersConfig[mapType].sources);
```
Comment (tltk90) [inline src/app/@ansyn/ansyn/modules/plugins/components/imagery-change-map/imagery-change-map.component.ts]: check if values is available

File: src/app/@ansyn/ol/mapSourceProviders/open-layers-ESRI-4326-source-provider.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ export const OpenLayerESRI_4326SourceProviderSourceType = 'ESRI_4326';
 })
 export class OpenLayerESRI4326SourceProvider extends OpenLayersMapSourceProvider<IESRI4326Config> {
 	create(metaData: IMapSettings): Promise<any> {
-		const { config } = this;
+		const config = {...this.config, ...metaData.data.config};
```
Comment (tltk90) [inline src/app/@ansyn/ol/mapSourceProviders/open-layers-ESRI-4326-source-provider.ts]: move merge to base

File: src/app/@ansyn/imagery/communicator-service/communicator.entity.ts [hunk 3]
```diff
@@ -95,6 +95,16 @@ export class CommunicatorEntity implements OnInit, OnDestroy {
 		return this.ActiveMap && this.ActiveMap.mapType;
 	}
 
+	public changeMapMainLayer(sourceType: string) {
+		this.mapSettings.worldView.sourceType = sourceType;
```
Comment (asafMasa) [inline src/app/@ansyn/imagery/communicator-service/communicator.entity.ts]: validate sourceType existance in sources

File: src/app/@ansyn/map-facade/public_api.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import { ChangeMainLayer } from './actions/map.actions';
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/public_api.ts]: remove import

File: src/app/@ansyn/ansyn/modules/plugins/components/imagery-change-map/imagery-change-map.component.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+			[attr.tooltip-value]="'Change map' | translate"
+			[attr.tooltip-class]="'left'">
+		<i class="icon icon-change-map"></i>
+	</button>
+</div>
+<div class="popup scrolled" *ngIf="showPopup">
+	<div *ngFor="let source of mapSources"
+		 class="map"
+	[class.active]="currentSourceType === source.key">
+		<button
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/plugins/components/imagery-change-map/imagery-change-map.component.html]: disable the button if currentSourceType === source.key

Comment (asafMasa) [reply src/app/@ansyn/ansyn/modules/plugins/components/imagery-change-map/imagery-change-map.component.html]: Great job 🥇

File: src/app/@ansyn/imagery/communicator-service/communicator.entity.ts [hunk 6]
```diff
@@ -96,13 +96,15 @@ export class CommunicatorEntity implements OnInit, OnDestroy {
 	}
 
 	public changeMapMainLayer(sourceType: string) {
```
Comment (asafMasa) [inline src/app/@ansyn/imagery/communicator-service/communicator.entity.ts]: return observable

File: src/app/@ansyn/imagery/communicator-service/communicator.entity.ts [hunk 7]
```diff
@@ -105,6 +105,7 @@ export class CommunicatorEntity implements OnInit, OnDestroy {
 				this.resetView(layer, position, [bbox[0], bbox[1], bbox[2], bbox[3]]);
 			})
 		}
+		return Promise.resolve(true);
```
Comment (asafMasa) [inline src/app/@ansyn/imagery/communicator-service/communicator.entity.ts]: need to return of(true)

**Pull request merged**